### DISCUSSION TITLE
Support diff view after inline assist

### DISCUSSION THREAD

**Discussion by SDGLBL** [olimorris/codecompanion.nvim#56] [Ideas]
<img width="1219" alt="image" src="https://github.com/olimorris/codecompanion.nvim/assets/41188874/f101c357-edc1-49e1-bb46-5428b237e686">

**Comment by olimorris** [comment]
Oh this is an awesome idea! I'll add this to the backlog

**Comment by olimorris** [comment]
![2024-06-29 11_23_39 - WezTerm@2x](https://github.com/olimorris/codecompanion.nvim/assets/9512444/42f9e1a9-6889-413f-a636-1bea1f5ceb71)

Initial attempt at this in https://github.com/olimorris/codecompanion.nvim/pull/58.

Thoughts so far:

- I don't think it's possible to have Tree-sitter highlights and a diff background (let me know if you've seen a plugin that has this)
- Given the above, I've left out extmarks for added code from the LLM

I've loosely tested this so would appreciate your feedback. I've changed some of the config defaults so you may need to adjust your settings.

**Reply by SDGLBL** [reply]
Thank you for sharing your update on adding diff highlighting to the plugin. I recall seeing a Neovim plugin called [diffview.nvim](https://github.com/sindrets/diffview.nvim) that was able to support both diff background colors and syntax highlighting simultaneously. I'm not certain if looking into how that plugin achieves this functionality would be helpful for your implementation, but I thought I would mention it in case it provides any useful insights or ideas.

<img width="1295" alt="image" src="https://github.com/olimorris/codecompanion.nvim/assets/41188874/5eafb7be-0a53-46b4-9864-b373174f086b">

**Comment by olimorris** [comment]
@SDGLBL it's now much improved:

![2024-07-01 17_20_06 - WezTerm@2x](https://github.com/olimorris/codecompanion.nvim/assets/9512444/e0049912-ed22-451a-a6d6-0bdc0b133d29)

Semantic tokens haven't been implemented as it's proven difficult to resolve the priority chain. Let me know what you think.

**Reply by olimorris** [reply]
I quite like the idea of adding some extmarks in the signcolumn which indicate where the plugin has generated code.

**Reply by SDGLBL** [reply]
I think the current results have already reached a usable state. Thank you very much.

**Comment by nuvic** [comment]
The diff view looks great! Thank you for making this :smile: 
I noticed that if there is no existing code (just a comment), then the diff will create a new buffer with the new code, instead of in the existing buffer.

Here is a test comment (lua) you can try:
```lua
-- Write a function that loops over a table
```

**Reply by olimorris** [reply]
Thanks for raising this. Can you share the exact steps e.g. the inline command and what text you select?

**Reply by nuvic** [reply]
Sure. This is some example text I've selected:
```
-- Write a function that loops over a table
```

And the command: `:'<,'>CodeCompanion Implement the following code as a senior software developer`
